### PR TITLE
Closes #137: Update Dashboard CRUD to handle groups.

### LINKED ISSUE #137: Update Dashboard CRUD to handle Groups

### PR DESCRIPTION
**Pull request by pml984**

### CODE REVIEW

File: routes/dashboards.js [hunk 1]
```diff
@@ -43,126 +32,122 @@ const roleAdmin = function (req, res, next) {
...
-      
+  const {group} = req.body
+  const {dashboard: dashboardId} = req.params
+ 
+  Dashboard
+    .findOneAndUpdate({'_id': req.dashboard._id}, req.body, {new: true})
+    .then(dashboard => utils.populateFullDashboard(dashboard).toObject())
+    .then(dashboard => {
+      dashboard.group = group
       res.json(dashboard)
```
Comment (jeffshaver) [inline routes/dashboards.js]: It seems confusing to send a response and return from the function

File: routes/dashboards.js [hunk 2]
```diff
@@ -43,126 +32,122 @@ const roleAdmin = function (req, res, next) {
...
+      
+      // Find dashboard within any of the dashboard groups.
+      return DashboardGroup
+        .findOne({dashboards: {$in: [dashboardId]}})
+    })
+    .then(prevDashboardGroup => {
+      if (group) {
```
Comment (jeffshaver) [inline routes/dashboards.js]: would look nicer if this was inverted

File: routes/dashboards.js [hunk 3]
```diff
@@ -43,126 +32,122 @@ const roleAdmin = function (req, res, next) {
...
+    .then(dashboard => utils.populateFullDashboard(dashboard).toObject())
+    .then(dashboard => {
+      dashboard.group = group
       res.json(dashboard)
+      
+      // Find dashboard within any of the dashboard groups.
+      return DashboardGroup
+        .findOne({dashboards: {$in: [dashboardId]}})
+    })
+    .then(prevDashboardGroup => {
```
Comment (jeffshaver) [inline routes/dashboards.js]: `prevDashboardGroup` is kind of a misleading name. maybe `existingDashbardGroup`?

File: routes/dashboard-groups.js [hunk 4]
```diff
@@ -1,74 +1,72 @@
+const mongoUtils = require('../utils/mongoUtil')
+const utils = require('../utils/dashboard')
 const express = require('express')
 const router = express.Router()
 
 module.exports = router
 
 const mongoose = require('mongoose')
+
+mongoose.Promise = require('bluebird')
```
Comment (jeffshaver) [inline routes/dashboard-groups.js:10]: Can't we just do this once inside on app.js to modify mongoose promises globally?

Comment (pml984) [reply routes/dashboard-groups.js:10]: Yeah, I saw you could do that in app.js, but I was going to change that once I switch all the routers to using promises.

File: routes/dashboards.js (hunk 1 above)
Comment (pml984) [reply routes/dashboards.js]: This appears to be standard for how mongoose works with express

**Pull request merged**